### PR TITLE
vte: respect the global systemdSupport flag

### PR DESCRIPTION
**Pull request by ghost**
This commit allows to compile vte without systemd (-D_systemd=false)
if the systemdSupport attribute is set to false.  This can be done
globally in nixpkgs.conf, and is understood by other packages.  See
also PR #160972.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/vte/default.nix [hunk 1]
```diff
@@ -45,6 +46,8 @@ stdenv.mkDerivation rec {
     })
   ];
 
+  mesonFlags = lib.optionals (!systemdSupport) [ "-D_systemd=false" ];
```
Comment (jtojnar) [inline pkgs/development/libraries/vte/default.nix:49]: Could you please move this under the inputs and put the list item on a separate line like with the inputs?

Comment (a-m-joseph) [reply pkgs/development/libraries/vte/default.nix]: Sorry about that!  Still learning the nixpkgs coding style.  Latest push should fix this.  Thanks for the prompt review!

Comment (jtojnar) [reply pkgs/development/libraries/vte/default.nix]: Sorry, by inputs I meant `propagatedBuildInputs` too.

It is actually not an official rule for Nixpkgs – it was observed as a common pattern in Nixpkgs, and some parts of the ecosystem (e.g. GNOME packaging) have stricter coding style that [incorporated it as a rule](https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/attribute-ordering.md). It is a bit unfortunate that we have differing coding styles throughout Nixpkgs but package sets like GNOME contain so many packages that such extra rules are necessary to make maintenance feasible with the limited amount of maintainers. At least for GNOME, you can use [nixpkgs-hammering](https://github.com/jtojnar/nixpkgs-hammering) to check that.

**Comment by a-m-joseph** [comment]
Thank you!

By the way, nixpkgs-hammer looks really neat.  I was going to run it on this PR before re-requesting a review from you, but its dependencies are building.  I have a pretty heavily customized setup so this happens a lot for me.

I have two PRs to submit for evince; since you mentioned that GNOME uses nixpkgs-hammer I will make sure those commits pass it before submitting the PR.

**Pull request merged**